src/dotnet/ToolPackage/ToolPackageInstaller.cs

```diff
@@ -112,10 +116,48 @@ internal class ToolPackageInstaller : IToolPackageInstaller
                 });
         }
 
+        public IToolPackage InstallPackageToExternalManagedLocation(PackageLocation packageLocation,
+            PackageId packageId,
+            VersionRange versionRange = null,
+            string targetFramework = null,
+            string verbosity = null)
+        {
+            var stageDirectory = _store.GetRandomStagingDirectory();
```

Comment:
So in `InstallPackage`, we do ensure the staging directory gets deleted when a rollback occurs.  Is that something we want for this method?

Reply:
Good catch. I realize this is not used, so I removed them.